### ISSUE TITLE
Progress bar for initial load

### ISSUE THREAD

**Issue by bballer4life** [albertobsd/keyhunt#18]
Hello @albertobsd, thank you for your software. It would be nice to see progress bar for initial load. Right now, I'm using k*bsgb mode with -k 1600, and it's running for an hour without starting or showing errors. My free system memory is equal to 6% all this time. I'm thinking, it's doing something in pre-allocated memory. But still it's only my predictions. The last thing displayed is: [+] precalculating 67... bP points

**Comment by dem10** [comment]
-k value like 120, it will use some 9 GB of RAM.You set the value to 1600, isn't it too high?

**Comment by bballer4life** [comment]
It's is related to old version, I have used -k 400 with old version.

**Comment by albertobsd** [comment]
@dem10 The actual version use less RAM, so you can use a bigger K value.

@bballer4life In my test with `-k 1024` load file take some 2 hours but the speed worth it!!

[https://albertobsd.dev/uploads/1616428067_bd1fc052-9441-4cbb-9bd1-d2e393489e18.jpg](image)

I can use ncurses to show % without add more lines, but this will increase the dependecies and complexity of the current code.

best regards!

**Comment by albertobsd** [comment]
But wait a momment, stop it, It will take some day or two in complete you are precalcualting the bP values not reading from a file if you want to read if from a file  (Having those values in a file is good if you plan to stop the process and start it again frecuently). Once you calculate it for one time, you only need to read the values from a file.

To create the file use bPfile liek

```
% ./bPfile
Create bP File usage
./bPfile <bP items> <output filename>

Example to create a File with 1 Billion items:
./bPfile 1000000000 Pfile.bin
```

The numeber depent of k value:

look:

```
albertobsd $ ./keyhunt -m bsgs -f tests/120.txt -k 1600 -b 120
[+] Version 0.1.20210322
[+] Setting mode BSGS
[+] Setting k factor to 1600
[+] Min range: 800000000000000000000000000000
[+] Max range: ffffffffffffffffffffffffffffff
[+] Opening file tests/120.txt
[+] Added 1 points from file
[+] Bit Range 120
[+] Setting N up to 17595944140800.
[+] Init 1st bloom filter for 6710886400 elements : 23004.00 MB
[+] Init 2nd bloom filter for 335544320 elements : 1150.00 MB
[+] Allocating 0.0 MB for 2622 aMP Points
[+] Precalculating 2622 aMP points
[+] Allocating 5120.00 MB for 335544320 bP Points
[+] precalculating 6710886400 bP points
^C
```

**[+] precalculating 6710886400 bP points**

the you need to run the file creation like:

```
% ./bPfile 6710886400 bPfile.bin
[+] precalculating 6710886400 bP elements in file bPfile.bin
```

and the load the file in keyhunt with ``-p ./bPfile.bin``

**Comment by albertobsd** [comment]
Solved in 37920b0000fdcfab30de2cae7fdf7ad032b89e23
